### ISSUE TITLE
EntitySerializer throws when using a MemoryStream created from a byte[]

### ISSUE THREAD

**Issue by Soren025** [friflo/Friflo.Engine.ECS#27]
The following code will throw an exception.

```csharp
EntityStore store = new();
Entity entity = store.CreateEntity();
entity.AddComponent(new EntityName { value = "Hello World!" });

EntitySerializer serializer = new();

using MemoryStream writeStream = new();
serializer.WriteEntities([entity], writeStream);

writeStream.Position = 0;
using StreamReader reader = new(writeStream);
string json = reader.ReadToEnd();

EntityStore newStore = new();
serializer.ReadIntoStore(newStore, new MemoryStream(Encoding.UTF8.GetBytes(json)));
```

The exception it will throw is...

```csharp
ERROR: System.UnauthorizedAccessException: MemoryStream's internal buffer cannot be accessed.
   at System.IO.MemoryStream.GetBuffer()
   at Friflo.Engine.ECS.Serialize.EntitySerializer.ReadIntoStoreSync(EntityStore store, MemoryStream memoryStream)
   at Friflo.Engine.ECS.Serialize.EntitySerializer.ReadIntoStore(EntityStore store, Stream stream)
   ...
```

There is a workaround if you construct the `MemoryStream` like so.

```csharp
byte[] bytes = Encoding.UTF8.GetBytes(json);
using MemoryStream memoryStream = new(bytes, 0, bytes.Length, true, true);
```

This works because the final `bool` parameter of the constructor being `true` opens up `GetBuffer()` to provide the internal `byte[]` instead of throwing an exception. However, I do not think this should be required.

I am using .NET 8 with version `3.0.0-preview.13` of `Friflo.Engine.ECS`.

**Comment by friflo** [comment]
@Soren025 Thx for report with test case.

Fixed with unit test.

**Comment by friflo** [comment]
Fixed in 3.0.0-preview.14
